subprojects/android-test/ui-testing-core-app/src/sharedTest/kotlin/com/avito/android/ui/test/ButtonsOverRecyclerTest.kt

```diff
@@ -1,13 +1,16 @@
 package com.avito.android.ui.test
 
 import androidx.test.espresso.action.SwipeDirections
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.avito.android.test.app.core.screenRule
 import com.avito.android.ui.ButtonsOverRecyclerActivity
 import org.hamcrest.Matchers.equalTo
 import org.hamcrest.Matchers.greaterThan
 import org.junit.Rule
 import org.junit.Test
+import org.junit.runner.RunWith
 
+@RunWith(AndroidJUnit4::class)
```

Comment:
Why do we need those? and what's the default runner?

Reply:
Espresso tests must run under instrumentation. `InstrumentationRegistry.getInstrumentation()` with throw an exception otherwise. [By default](https://cs.android.com/androidx/android-test/+/master:ext/junit/java/androidx/test/ext/junit/runners/AndroidJUnit4.java;l=59), `RobolectricTestRunner` is used for unit tests and `AndroidJUnit4ClassRunner` for Android tests on the emulator (if it is not overridden by `defaultConfig.testInstrumentationRunner`).

Reply:
Could it be set globally for sharedTest somehow?
It's easy to forget to add one, and i bet an error will be hard to debug

Reply:
Didn't find a way to do that. 

You can face the problem in 2 cases: 
1. When creating a new test
2. When moving the existing test to the shared folder

In both cases the error message "`java.lang.IllegalStateException: No instrumentation registered! Must run under a registering instrumentation.`" seems understandable and clear.

Reply:
> seems understandable and clear

I can't see how it's connected with run annotation